lighthouse/helpers/plate_events.py

```diff
@@ -171,7 +128,45 @@ def construct_source_plate_no_map_data_message(
         {[str]} -- Any errors attempting to construct the message, otherwise an empty array.
         {Message} -- The constructed message; otherwise None if there are any errors.
     """
-    return __construct_default_source_plate_on_robot_message(PLATE_EVENT_SOURCE_NO_MAP_DATA, params)
+    try:
+        barcode = params.get("barcode", "")
+        user_id = params.get("user_id", "")
+        robot_serial_number = params.get("robot", "")
+        if len(barcode) == 0 or len(user_id) == 0 or len(robot_serial_number) == 0:
```

Comment:
Can just do `if not barcode or ...` instead.

Reply:
:+1: learning more "pythonic" approaches each day